pkg/cli/cli.go

```diff
@@ -1423,6 +1424,20 @@ func runConnectionWebSocketAgentCommand(flags *ConnectFlags, c *command, args []
 		cerr.Close()
 	}
 
+	go func() {
+		ticker := time.NewTicker(10 * time.Second)
+		defer ticker.Stop()
+		for {
+			select {
+			case <-ticker.C:
+				if err := wsConn.WriteControl(websocket.PingMessage, []byte{}, time.Now().Add(5*time.Second)); err != nil {
+					c.PrintErrf("Failed to write Ping: %v", err)
+					return
```

Comment:
IIUC this will only exit when there is an error. There should be a mechanism to do an orderly shutdown of this routine. For example, there could be another channel in the `select` call that simply causes it to return on any message received. The channel would be written to right before closing the `wsWrapper` in line 1450.

Reply:
Added channel for the scenario that socket is shutdown normally.

Moved go routine code into newWsIoWrapper function that is currently introduced for create wsIoWrapper.